pkg/controllers/deprovisioning/helpers.go

```diff
@@ -325,13 +325,13 @@ func mapNodes(nodes []*v1.Node, candidateNodes []CandidateNode) []CandidateNode
 	return ret
 }
 
-func canBeTerminated(node CandidateNode, pdbs *PDBLimits) (string, bool) {
+func canBeTerminated(node CandidateNode, pdbs *PDBLimits) (string, bool, bool) {
```

Comment:
I feel like if we are getting this specific with this many return args, maybe there's no use in this helper func anymore. Thoughts?

Reply:
Or at least it would be good to describe the returned values

Reply:
I think it helps to have it all within one function. Maybe we could represent the reason better than a string, but I don't see this as too complicated.

Reply:
It's less about how complicated it is and more if there is any value in abstracting a concept. If you are having to return a bunch of different things to represent why something failed, you are no longer abstracting the concept. We have had to create a check for `pdbs == nil` because emptiness uses this and has pdbs, now we are trying to surface whether pods have DoNotEvict for the node.

I think I'd just rather see

```go
if !node.Deletion.TimeStamp.IsZero() {
}
if !pdbs.CanEvictPods(node.pods) {
}
if HasDoNotEvict(node.pods) {
}
```

Then, you can handle the event firing in the `if` statement